airbyte-integrations/connectors/source-klaviyo/source_klaviyo/streams.py

```diff
@@ -0,0 +1,173 @@
+                If there are no more pages in the result, return None.
+        """
+        decoded_response = response.json()
+        if decoded_response["end"] < decoded_response["total"] - 1:  # end is zero based
+            return {
+                "page": decoded_response["page"] + 1,
+            }
+
+        return None
+
+    def request_params(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+        """Usually contains common params e.g. pagination size etc."""
+        next_page_token = next_page_token or {}
+        return {**next_page_token, "api_key": self._api_key, "count": self.page_size}
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        """
+        :return an iterable containing each record in the response
+        """
+        response_json = response.json()
+        for record in response_json.get("data", []):  # Stripe puts records in a container array "data"
+            self.schema.parse_obj(record)  # validate with schema first
+            yield record
+
+    def get_json_schema(self) -> Mapping[str, Any]:
+        """Use Pydantic schema"""
+        return self.schema.schema()
+
+
+class Campaigns(KlaviyoStream):
+    schema = Campaign
+
+    def path(self, **kwargs) -> str:
+        return "campaigns"
+
+
+class Lists(KlaviyoStream):
+    schema = PersonList
+
+    def path(self, **kwargs) -> str:
+        return "lists"
+
+
+class GlobalExclusions(KlaviyoStream):
+    schema = GlobalExclusion
+    primary_key = "email"
+
+    def path(self, **kwargs) -> str:
+        return "people/exclusions"
+
+
+class Metrics(KlaviyoStream):
+    schema = Metric
+
+    def path(self, **kwargs) -> str:
+        return "metrics"
+
+
+class IncrementalKlaviyoStream(KlaviyoStream, ABC):
+    state_checkpoint_interval = 100
+
+    def __init__(self, start_date: str, **kwargs):
+        super().__init__(**kwargs)
+        self._start_ts = int(pendulum.parse(start_date).timestamp())
+
+    @property
+    @abstractmethod
+    def cursor_field(self) -> str:
+        """
+        Override to return the cursor field used by this stream e.g: an API entity might always use created_at as the cursor field. This is
+        usually id or date based. This field's presence tells the framework this in an incremental stream. Required for incremental.
+
+        :return str: The name of the cursor field.
+        """
+
+    def request_params(self, stream_state=None, **kwargs):
+        stream_state = stream_state or {}
+        params = super().request_params(stream_state=stream_state, **kwargs)
+
+        if not params.get("since"):  # skip state filter if already have one from pagination
+            params["since"] = max(stream_state.get(self.cursor_field, 0), self._start_ts)
```

Comment:
do we need to wrap `.get` with an `int()` call?

Reply:
assuming we control state this should not be necessary, but again if user will create it manually we probably can save few keys for him